accounts/models.py

```diff
@@ -358,6 +358,12 @@ def delete_user(self, remove_sounds=False):
         else:
             Sound.objects.filter(user=self.user).update(is_index_dirty=True)
 
+    def has_content(self):
+        return Sound.objects.filter(user=self.user) or \
+               Post.objects.filter(author=self.user) or \
+               Comment.objects.filter(user=self.user)
+        # TODO: add more checks?
```

Comment:
Yes we should add more checks:

- Packs
- Ratings
- Downloads
- PackDownloads
- Bookmarks
- APIV2 clients

Can we list all tables which have foreign keys to user_id to make sure we don't miss any important thing?
Then we can choose which things are "important" and should be included and which not. Or maybe we decide to include them all. In any case this should be properly documented in `has_content`.

Reply:
Current list of tables that are referencing user:
- `accounts_emailbounce`
- `accounts_oldusername`
- `accounts_profile`
- `accounts_resetemailrequest`
- `accounts_sameuser`
- `accounts_sameuser`
- `accounts_useremailsetting`
- `apiv2_extendedapplication`
- `auth_message`
- `comments_comment`
- `django_admin_log`
- `donations_donation`
- `favorites_favorite`
- `forum_post`
- `forum_subscription`
- `forum_thread`
- `fsmessages_message`
- `fsmessages_message`
- `general_akismetspam`
- `geotags_geotag`
- `oauth2_provider_accesstoken`
- `oauth2_provider_application`
- `oauth2_provider_grant`
- `oauth2_provider_refreshtoken`

Reply:
I updated the list of "important" things I made with info from here. You removed some things from this list right? There is no `sounds_sound` table but I assume you removed it because it was already being considered right?

Reply:
My bad, I haven't included all of them. Now updated the list

Reply:
Thanks, I edited again my list, I think checking these objects should be enough to consider user has "important" content.